src/app/staking/[addressOrENSName]/deposits/Deposit.tsx

```diff
@@ -0,0 +1,189 @@
+import { useQueryClient } from "@tanstack/react-query";
+
+import { TOKEN_BALANCE_QK, useTokenBalance } from "@/hooks/useTokenBalance";
+import { DEPOSITOR_TOTAL_STAKED_QK } from "@/hooks/useDepositorTotalStaked";
+import { INDEXER_DELAY } from "@/lib/constants";
+import { TOKEN_ALLOWANCE_QK } from "@/hooks/useTokenAllowance";
+
+interface DepositProps {
+  deposit: StakedDeposit;
+  fetchDelegate: (address: string) => Promise<Delegate>;
+  refreshPath: (path: string) => void;
+}
+
+export const Deposit = ({
+  deposit,
+  fetchDelegate,
+  refreshPath,
+}: DepositProps) => {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+  const { isConnected, address } = useAccount();
+  const { data: tokenBalance } = useTokenBalance(address);
+  const [delegate, setDelegate] = useState<Delegate | null>(null);
+  const isDelegateFetched = useRef(false);
+
+  const { contracts } = Tenant.current();
+  const { config } = usePrepareContractWrite({
+    address: contracts.staker!.address as `0x${string}`,
+    abi: contracts.staker!.abi,
+    chainId: contracts.staker!.chain.id,
+    functionName: "withdraw",
+    args: [BigInt(deposit.id), BigInt(deposit.amount)],
+  });
+
+  const { data, write } = useContractWrite(config);
+  const isDepositOwner =
+    isConnected && address?.toLowerCase() === deposit.depositor.toLowerCase();
+
+  const { isLoading: isProcessingWithdrawal, isFetched: didProcessWithdrawal } =
+    useWaitForTransaction({
+      hash: data?.hash,
+    });
+
+  const getDelegate = async () => {
+    if (!isDelegateFetched.current) {
+      const delegate = await fetchDelegate(deposit.delegatee as `0x${string}`);
+      setDelegate(delegate);
+      isDelegateFetched.current = true;
+    }
+  };
+
+  useEffect(() => {
+    //  Fetch delegate if not already fetched
+    if (!delegate && !isDelegateFetched.current) {
+      getDelegate();
+    }
+    // Refresh route and invalidate cache if withdrawal was processed
+    if (didProcessWithdrawal) {
+      setTimeout(() => {
+        Promise.all([
+          queryClient.invalidateQueries({
+            queryKey: [TOKEN_BALANCE_QK, deposit.depositor],
+          }),
+          queryClient.invalidateQueries({
+            queryKey: [DEPOSITOR_TOTAL_STAKED_QK, deposit.depositor],
+          }),
+          queryClient.invalidateQueries({
+            queryKey: [TOKEN_ALLOWANCE_QK],
+          }),
+        ]).then(() => {
+          refreshPath(`/staking/${deposit.depositor}`);
+          router.refresh();
+        });
+      }, INDEXER_DELAY);
+    }
+  }, [
+    delegate,
+    deposit,
+    didProcessWithdrawal,
+    getDelegate,
+    isDelegateFetched,
+    queryClient,
+    router,
+  ]);
+
+  return (
+    <div className="px-5 py-4 w-full">
+      <div className="flex flex-row gap-5 justify-between w-full">
+        <div className="w-[130px] border-r border-gray-300">
+          <div className="text-xs font-medium text-gray-700">Staked</div>
```

Comment:
we definitely have a lot of colors lol I've been using `text-agora-stone-700` which is a new set of colors I added to tailwind based off of the figma mocks. Eventually we should do a pass and standardize all the colors. 

(will likely have to happen when we move to customizing tenant colors)